Themes/default/GenericControls.template.php

```diff
@@ -154,7 +154,25 @@ function template_control_richedit_buttons($editor_id)
 	elseif ($context['show_spellchecking'])
 		$tempTab++;
 
-	$tempTab++;
+	// add a cancel button for these actions
+	if(isset($_REQUEST['action']) && in_array($_REQUEST['action'], array('post', 'calendar')))
+	{
+		$cancelLink = '';
+		if(isset($_REQUEST['msg']) || isset($_REQUEST['last_msg']))
+			$cancelLink = $scripturl . '?topic='. $_REQUEST['topic'] .'.msg'. (isset($_REQUEST['msg']) ? $_REQUEST['msg'] .'#msg'. $_REQUEST['msg'] : $_REQUEST['last_msg'] .'#msg'. $_REQUEST['last_msg']);
+		else if(isset($_REQUEST['board']))
```

Comment:
SMF uses `elseif` instead of `else if` unfortunately. Also space here and on all other ifs.